commander3/src/comm_nonlin_mod.f90

```diff
@@ -3241,9 +3899,76 @@ subroutine sampleDiffuseSpecIndPixReg_nonlin(cpar, buffer_lnL, handle, comp_id,
     call theta_single_fr%dealloc(); deallocate(theta_single_fr); theta_single_fr => null()
     call mask_lr%dealloc();         deallocate(mask_lr);         mask_lr => null()
     call res_map%dealloc();         deallocate(res_map);         res_map => null()
+    
+    if (c_lnL%spec_mono_combined(par_id)) then
+       call mask_mono%dealloc() 
+       deallocate(mask_mono)
+       mask_mono => null()
+    end if
+
+    if (allocated(monopole_val)) deallocate(monopole_val)
+    if (allocated(monopole_mu)) deallocate(monopole_mu)
+    if (allocated(monopole_rms)) deallocate(monopole_rms)
+    if (allocated(monopole_mixing)) deallocate(monopole_mixing)
+    if (allocated(monopole_active)) deallocate(monopole_active)
+    if (allocated(old_mono)) deallocate(old_mono)
+    if (allocated(new_mono)) deallocate(new_mono)
+
 
   end subroutine sampleDiffuseSpecIndPixReg_nonlin
 
+  function calc_corr_coeff(spec_arr, n_spec, n_lim)
```

Comment:
This should probably be moved to math_mod or utils or something like that

Reply:
I can do that